routes/payment_intent/index.mjs

```diff
@@ -0,0 +1,297 @@
+//@ts-check
+import { Hono } from "hono";
+import { createSupabaseClient } from "../../utils/supabase.mjs";
+const app = new Hono();
+
+// Initialize Supabase client
+const supabase = createSupabaseClient();
+
+//create payment intent
+app.post("/", async (c) => {
+  const { from_number } = await c.req.json().catch(() => ({}));
+
+  if (!from_number) {
+    return c.json(
+      {
+        message: "Invalid request; missing `from_number`",
+      },
+      400
+    );
+  }
+
+  try {
+    // @ts-ignore
+    const { data, error } = await supabase
+      .from("intents")
+      .insert({
+        from_number: from_number,
+        client_secret: `client_secret_${from_number}`,
+      })
+      .select();
+    if (error) {
+      console.log(error);
+    }
+
+    return c.json(data, 201);
+  } catch (err) {
+    console.error("Unexpected error:", err);
+    return c.json(
+      {
+        message: "Unexpected error occurred",
+        error: err.message,
+      },
+      500
+    );
+  }
+});
+
+//update payment intent
+app.put("/:id", async (c) => {
+  const id = c.req.param("id");
+  const {
+    from_number,
+    to_number,
+    amount,
+    currency,
+    description,
+    cancelation_reason,
+    payment_method,
+    amount_received,
+  } = await c.req.json().catch(() => ({}));
+  const secret = `client_secret_${from_number}`;
+
+  try {
+    const { data, error } = await supabase
+      .from("intents")
+      .update({
+        from_number: from_number,
+        to_number: to_number,
+        amount: amount,
+        currency: currency,
+        description: description,
+        cancelation_reason: cancelation_reason,
+        payment_method: payment_method,
+        amount_received: amount_received,
```

Comment:
Im not sure if this works, since we will need the ability to partially update the intent.

e.g.
When we get the amount, we update the payment intent with just the amount

Reply:
It works as intended. the fields are nullable. it will only update the fields that are not empty

it is flexible enough to allow updating of the intent as the client progress with their transaction